app/src/main/java/com/epfl/drawyourpath/database/FireDatabase.kt

```diff
@@ -38,76 +70,175 @@ class FireDatabase : Database() {
+            .addOnFailureListener { future.completeExceptionally(java.lang.Error("Impossible to set the distance goal to the database"))}
+        return future
     }
 
-    override fun setActivityTimeGoal(userId: String, activityTimeGoal: Double): CompletableFuture<Boolean> {
-        TODO("Will be done in the next task: database rebase")
+    override fun setActivityTimeGoal(activityTimeGoal: Double): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        val dataUpdated = HashMap<String, Any>()
+        dataUpdated.put("activityTimeGoal", activityTimeGoal)
+
+        if(activityTimeGoal <= 0.0){
+            throw java.lang.Error("The activity time goal can't be less or equal than 0.")
+        }
+
+        database.child(usersProfileFileName).child(getUserId()).updateChildren(dataUpdated)
+            .addOnSuccessListener { future.complete(true) }
+            .addOnFailureListener { future.completeExceptionally(java.lang.Error("Impossible to set the activity time goal to the database"))}
+        return future
     }
 
-    override fun setNbOfPathsGoal(userId: String, nbOfPathsGoal: Int): CompletableFuture<Boolean> {
-        TODO("Will be done in the next task: database rebase")
+    override fun setNbOfPathsGoal(nbOfPathsGoal: Int): CompletableFuture<Boolean> {
+        val future = CompletableFuture<Boolean>()
+        val dataUpdated = HashMap<String, Any>()
+        dataUpdated.put("nbOfPathsGoal", nbOfPathsGoal)
+
+        if(nbOfPathsGoal <= 0.0){
+            throw java.lang.Error("The number of paths goal can't be less or equal than 0.")
+        }
+
+        database.child(usersProfileFileName).child(getUserId()).updateChildren(dataUpdated)
+            .addOnSuccessListener { future.complete(true) }
+            .addOnFailureListener { future.completeExceptionally(java.lang.Error("Impossible to set the number of paths goal to the database"))}
+        return future
     }
 
     /**
-     * Helper functions to add some data to a user profile withe his username to the database
-     * @param data date to add to the database
-     * @param username corresponding to the user profile
+     * Helper function to get the userId from the authentication and check if a user is log
+     * @return the userId of the user log on the app
+     * @throw an error if any user is log on the app
      */
-    private fun updateUserData(data: Map<String, Any>, username: String) {
-        //TODO:Fix this function in the next task such that the user profile is associated to the userId
-        database.child("users").child(username).updateChildren(data)
-            .addOnFailureListener {
-                throw Exception(buildString {
-                    append("Impossible to add the data on the user: ")
-                    append(username)
-                    append(" on the database")
-                })
-            }
+    private fun getUserId(): String{
+        if(userAuth == null){
```

Comment:
Throwing an exception is maybe a bit overkill here. Couldn't we return `null`?

Reply:
We couldn't because we need a non null userid in the functions that that call this function.

Reply:
I see what you mean, but there are two main problem: the user id is fetched upon the creation of the UserModel which means that we could reference a stale uid. All methods using this function are asynchronous and return a future. If this function starts throwing, the exception won't be forwarded through the future and will just crash the app.